File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
// maxDepth is going to be max open brackets at any time
public class Solution {
    public int maxDepth(String s) {
        int maxDepth = 0, openBrackets = 0;
        
        for(var ch : s.toCharArray()){
            if(ch == '(') openBrackets++;
            else if(ch == ')') openBrackets--;

            maxDepth = Math.max(maxDepth, openBrackets);
        }

        return maxDepth;
    }
}

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.py
# maxDepth is going to be max open brackets at any time
class Solution:
    def maxDepth(self, s: str) -> int:
        max_depth = open_brackets = 0

        for ch in s:
            if ch == '(':
                open_brackets += 1;
                max_depth = max(max_depth, open_brackets)
            elif ch == ')':
                open_brackets -= 1;
        
        return max_depth